### PR TITLE
PLU-329: allow cc email address for postman

### PR DESCRIPTION
**Pull request by kevinkim-ogp**
## Problem
Email by Postman does not support CC.

## Solution
Add CC recipients field.

## Before & After Screenshots
**BEFORE**:
Screenshot 2024-11-26 at 11 58 52 AM

**AFTER**:
Screenshot 2024-11-26 at 12 01 38 PM

## Tests
- [x] Existing email functionality works
- [x] Emails can be sent without CC recipients
- [x] Emails can be sent to 1 or many CC recipients

### CODE REVIEW

File: packages/backend/src/apps/postman/common/parameters.ts [hunk 1]
```diff
@@ -15,6 +15,17 @@ function recipientStringToArray(value: string) {
...
+    ctx.addIssue({
+      code: z.ZodIssueCode.custom,
+      message: msg,
+    })
+  }
+  return recipients
+}
+
 export const transactionalEmailFields: IField[] = [
   {
```
Comment (liangyuanruo) [inline packages/backend/src/apps/postman/common/parameters.ts:30]: LGTM!

**Review by liangyuanruo — APPROVED** [review]

File: packages/backend/src/apps/postman/common/parameters.ts [hunk 2]
```diff
@@ -39,6 +50,15 @@ export const transactionalEmailFields: IField[] = [
       'To send to multiple recipients, comma-separate email addresses. Emails will be sent to each email address separately.',
```
Comment (pregnantboy) [inline packages/backend/src/apps/postman/common/parameters.ts]: ChatGPT refined:
```suggestion
      'Enter the email addresses of the main recipients, separated by commas. Each recipient will receive an individual email.',
```

File: packages/backend/src/apps/postman/common/parameters.ts [hunk 3]
```diff
@@ -39,6 +50,15 @@ export const transactionalEmailFields: IField[] = [
...
     variables: true,
   },
+  {
+    label: 'CC recipient email(s)',
+    key: 'destinationEmailCc',
+    type: 'string' as const,
+    required: false,
+    description:
+      'To CC multiple recipients, comma-separate email addresses. CC recipients will receive multiple copies of the email.',
```
Comment (pregnantboy) [inline packages/backend/src/apps/postman/common/parameters.ts]:
```suggestion
      'Enter the email addresses to CC, separated by commas. CC recipients will receive a copy of the email for each main recipient.',
```

File: packages/backend/src/apps/postman/common/email-helper.ts [hunk 4]
```diff
@@ -89,6 +90,13 @@ export async function sendTransactionalEmails(
     )
     requestData.append('disable_tracking', 'true')
 
+    if (email.ccList?.length) {
+      // For single CC, send as JSON array. For multiple CCs, append individually
+      email.ccList.length === 1
+        ? requestData.append('cc', JSON.stringify(email.ccList))
+        : email.ccList.forEach((value) => requestData.append('cc', value))
```
Comment (pregnantboy) [inline packages/backend/src/apps/postman/common/email-helper.ts]: as discussed, just stringify

**Review by pregnantboy — APPROVED** [review]
nits

**Pull request merged**